Prepare for 3.0.3 release

matthijskooijman opened the pull request:
This PR starts adding some release notes for previously merged. This replaces #614, but this PR is created from a branch in *this* repository, making it a bit easier for all maintainers to update it as we go along.

As for release planning: I'd like to move to a 3.0.3 release, since we've already collected some relevant bugfixes in master. I would like to at least also merge #630 before releasing, which fixes the worst usability issues in the edit activity screen (but needs a bit more work, probably from my side).

Any other things that we would want to get into this release? I looked through the open PRs, but most of the recent ones still need more work, and the older ones need a better looking-at. #631 might be easy, but I haven't taken the time to review it yet (and I probably won't want to delay release for it).

GeraldJansen commented:
It is very encouraging to see the recent activity and the approaching 3.03 release. Concerning 630, even a partial fix would be a worthwhile improvement.

rhertzog commented:
I would love to see some discussion on https://github.com/projecthamster/hamster-shell-extension/issues/334 and see wether we prefer to revert 42d3c05 here or find some other solution elsewhere.

matthijskooijman commented:
> I would love to see some discussion on projecthamster/hamster-shell-extension#334 and see wether we prefer to revert 42d3c05 here or find some other solution elsewhere.

Good point, I had missed that. I created a separate issue for that: https://github.com/projecthamster/hamster/issues/657

matthijskooijman commented:
I've rebased and added changelog entries for recently merged PRs. I looked at #631, but I think it's not near to being merged, so I think the last remaining PR to consider for this release would be the double-comma one (#663).

rabin-io commented:
Hi @matthijskooijman what about small PR like #690 ?
can you marge it ?

lynxis commented:
I think everything is still ready for a next release.
The following 2 issues/prs would also be great to have a new release to allow hamster running on py 3.10.
https://github.com/projecthamster/hamster/pull/715
and https://github.com/projecthamster/hamster/issues/721

ma-adrian commented:
Some distros like Manjaro have already updated to python versions where the 3.0.2 release is not starting anymore (#726). I think it would be time to make a new release to fix this issue.

matthijskooijman commented:
I'm going to close this for now, and then make a new PR with a clean history for the final release.
